src/home_robot/home_robot/motion/stretch.py

```diff
@@ -184,26 +187,41 @@ class HelloStretchKinematics(Robot):
         "joint_wrist_roll",
     ]
 
-    def _create_ik_solvers(self, ik_type: str = "pybullet", visualize: bool = False):
+    def _create_ik_solvers(
+        self, ik_type: str = "pybullet", visualize: bool = False
+    ):  # TODO: spowers unused visualize flag?
         """Create ik solvers using pybullet or something else."""
         # You can set one of the visualize flags to true to debug IK issues
         # This is not exposed manually - only one though or it will fail
-        if ik_type == "pybullet":
+        assert ik_type in [
+            "pybullet",
+            "pinocchio",
+            "pybullet_optimize",
+            "pinocchio_optimize",
+        ], f"Unknown ik type: {ik_type}"
+
+        # You can set one of the visualize flags to true to debug IK issues
+        if "pybullet" in ik_type:
```

Comment:
I admit this "in" pattern is a little strange here, but ... it is very convenient, as it allows composition of the base method with the optimizer in an easy way. Open to alternatives though.

Reply:
One potentially cleaner way is to have a factory pattern that generates the IK solver object based on two params - `ik_type: str` and `optimize_orientation: bool`.

Reply:
Yeah, I considered that. TBH I went with this way because it was a smaller change. Won't be offended if you change it, though.